File: app/src/main/java/com/parabola/newtone/mvp/view/TabTrackView.java
package com.parabola.newtone.mvp.view;

import com.parabola.domain.model.Track;
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView;

import java.util.List;

import moxy.MvpView;
import moxy.viewstate.strategy.alias.AddToEndSingle;

@AddToEndSingle
public interface TabTrackView extends MvpView {

    void refreshTracks(List<Track> tracks);
    void setItemViewSettings(TrackItemView trackItemViewSettings);
    void setItemDividerShowing(boolean showed);

    void setCurrentTrack(int trackId);
    void setSectionShowing(boolean enable);

    void removeTrack(int trackId);
}

File: app/src/main/java/com/parabola/newtone/mvp/view/TabTrackView.kt
package com.parabola.newtone.mvp.view

import com.parabola.domain.model.Track
import com.parabola.domain.settings.ViewSettingsInteractor.TrackItemView
import moxy.MvpView
import moxy.viewstate.strategy.alias.AddToEndSingle

@AddToEndSingle
interface TabTrackView : MvpView {
    fun setCurrentTrack(trackId: Int)
    fun refreshTracks(tracks: List<Track>)

    fun setItemViewSettings(viewSettings: TrackItemView)
    fun setItemDividerShowing(showed: Boolean)

    fun setSectionShowing(enable: Boolean)

    fun removeTrack(trackId: Int)
}
